railties/lib/rails/application/configuration.rb

```diff
@@ -232,6 +233,7 @@ def load_defaults(target_version)
           if respond_to?(:active_record)
             active_record.verify_foreign_keys_for_fixtures = true
             active_record.partial_inserts = false
+            active_record.encryption.hash_digest_class = OpenSSL::Digest::SHA256
```

Comment:
@jorgemanrubia Maybe a dumb question, but do we also need to update `railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_0.rb.tt`?

I'm not sure we should be spending too much time on super-early adopters. Yes, they will have to maintain SHA1 for backwards compatibility if they're using Rails 7 with Rails 6 defaults (e.g. Hey), but are we over-complicating the upgrade path to Rails 7 unnecessarily for a very small minority?

Reply:
@rmm5t Good call, thanks! I added an entry to `new_framework_defaults_7_0`.

> I'm not sure we should be spending too much time on super-early adopters. 

I am concerned that the consequences are too serious for people running edge-Rails. Suddenly, you upgrade to latest `main` and your existing production data fails to decrypt. It can be super-confusing, so I am happy to put the extra-effort here. I think it's a _small_ extra effort 😅.